lib/heap_sort.rb

```diff
@@ -1,8 +1,84 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
-end
\ No newline at end of file
+# Time Complexity: O(nlogn)??
+# Space Complexity: O(1)
+def heapsort(list)
```

Comment:
Nice work at attempting an in-place heapsort.  I like the code you wrote with good helper methods.  One thing.  Since you're doing recursion, the space complexity is O(log n).  So it's better than MergeSort, but not quite O(1).